lib/fs.js

```diff
@@ -56,7 +56,7 @@ function getOptions(options, defaultOptions) {
 
   if (options.encoding !== 'buffer')
     assertEncoding(options.encoding);
-  return options;
+  return Object.assign({}, defaultOptions, options);
```

Comment:
never done a benchmark. do we have any existing ones to compare it with? or any suggestions how to do a benchmark? Thanks